src/gst-plugins/webrtcendpoint/kmswebrtctransportsrcnice.c

```diff
@@ -52,17 +209,93 @@ kms_webrtc_transport_src_nice_configure (KmsWebrtcTransportSrc * self,
   g_object_set (G_OBJECT (self->src),
       "agent", kms_ice_nice_agent_get_agent (nice_agent),
       "stream", id, "component", component_id, NULL);
+
+  g_signal_connect (nice_agent, "on-ice-component-state-changed",
+    G_CALLBACK (kms_webrtc_transport_src_nice_component_state_changed), self);
+}
+
+// Store DTLS buffers for later delivery when ICE gets to CONNECTED.
+static GstPadProbeReturn
+kms_webrtc_transport_src_nice_block_dtls_until_ice_connected (GstPad *pad, GstPadProbeInfo *info, gpointer user_data)
+{
+  KmsWebrtcTransportSrcNice *self = KMS_WEBRTC_TRANSPORT_SRC_NICE(user_data);
+  GstPadProbeReturn ret = GST_PAD_PROBE_OK;
+
+  KMS_WEBRTC_TRANSPORT_SRC_NICE_LOCK (self);
+  if (self->priv->pending_buffers_delivered) {
+    KMS_WEBRTC_TRANSPORT_SRC_NICE_UNLOCK (self);
+    GST_DEBUG_OBJECT (self, "No more possible buffers pending, removing probe");
+    return GST_PAD_PROBE_REMOVE;
+  }
+
+  if (GST_PAD_PROBE_INFO_TYPE(info) & GST_PAD_PROBE_TYPE_BUFFER) {
+    GstBuffer *buffer = gst_pad_probe_info_get_buffer (info);
+
+    if (buffer != NULL) {
+      gst_buffer_ref (buffer);
```

Comment:
As mentioned: No need to ref here, as long as no unref is done in `store_pending_dtls_buffer`